storagemarket/types.go

```diff
@@ -277,6 +277,9 @@ type StorageDeal struct {
 	market.DealState
 }
 
+// Subscriber is a callback that is called when events are emitted
+type ProviderSubscriber func(event ProviderEvent, deal *MinerDeal)
```

Comment:
this should not be a pointer. You cannot modify the miner deal, so pass the value.